src/c_api.cc

```
@@ -58,6 +58,28 @@ class ReadCbStreamBuf : public std::streambuf {
+// It is not free. The parallel path allocates one part buffer per in-flight
+// part and registers each for RDMA, so the pinned memory is
+// max_inflight_parts * part_size -- 64 MiB at this default. A caller running
+// many concurrent uploads multiplies that, which is why the default stays
+// modest and tuning is left to the environment rather than raised for
+// everyone.
+inline unsigned int StreamInflightParts() {
+  if (const char* env = std::getenv("MINIOCPP_STREAM_INFLIGHT_PARTS")) {
+    char* end = nullptr;
+    const unsigned long v = std::strtoul(env, &end, 10);
+    if (end != env && v >= 1 && v <= 100) return static_cast<unsigned int>(v);
+  }
```

Cppcheck (2.21.0)

[style] The function 'GetBaseUrl' is never used.

(unusedFunction)